### PR TITLE
Some math additions.

### PR DESCRIPTION
**Pull request by Nexela**
See changelog for a list.

### CODE REVIEW

**Comment by raiguard** [comment]
Honestly, I like the distinction between `round` and `round_to` more than combining the functions. It makes it very clear that you're using one or the other, whereas `rounded` is much less clear. Same applies for `ceil` and `floor`.

Additionally `degrees` and `radians` are much less clear than `rad_to_deg` and `deg_to_rad`.

The tests are pretty cool though!

**Comment by Nexela** [comment]
Round is still the same round.

ceiled/floored are open to better names.  I would have used ceil_to/floor_to  but those are used by the more ineffiecent versions and would be a breaking change for anyone using them. If it wasn't so much overhead involved the best thing to do would be to swap ceil/floor.

I will probably throw the tests back up once I get it to run reliably without a janky setup (I think I am almost there) and can get some setup instructions going.  I was going to go with busted as it is a much better suite but luaunit has a mostly working vscode sidebar plugin...... and the in depthness of busted probably isn't needed.

File: math.lua [hunk 1]
```diff
@@ -75,14 +51,29 @@ function flib_math.round(num, divisor)
   end
 end
 
+--- Ceil a number to N decimal places.
```
Comment (raiguard) [inline math.lua]: These functions don't round to N decimal places, they round to the nearest multiple of N.

File: math.lua [hunk 2]
```diff
@@ -75,14 +51,29 @@ function flib_math.round(num, divisor)
...
+--- @param num number
+--- @param divisor? number `num` will be rounded to the nearest multiple of `divisor` (default: 1).
+function flib_math.ceiled(num, divisor)
+  if divisor then return flib_math.ceil(num / divisor) * divisor end
+  return flib_math.ceil(num)
+end
+
+--- Floor a number to N decimal places.
```
Comment (raiguard) [inline math.lua]: Same here - this is inaccurate.

File: math.lua [hunk 3]
```diff
@@ -146,4 +179,8 @@ function flib_math.lerp(num1, num2, amount)
   return num1 + (num2 - num1) * flib_math.clamp(amount, 0, 1)
 end
 
+--- The number of ticks in a real unit of time.
+--- @return number
+flib_math.time = { second = 60, minute = 3600, hour = 216000, day = 5184000 }
```
Comment (raiguard) [inline math.lua]: I don't really like having these in `math`. Part of my plan is to create a `time` module and move `flib_misc.ticks_to_timestring` into it. These would be good values to have in there as well.

**Review by raiguard — CHANGES_REQUESTED** [review]
Overall, some pretty good additions! Thanks for the clarification on ceil/floor to - I originally though you had just renamed the functions. If they have different functionality then that's fine.

Please run `stylua` on the file as well.

File: math.lua (hunk 1 above)
Comment (Nexela) [reply math.lua]: CopyPasta FTW

**Review by raiguard — APPROVED** [review]

**Pull request merged**